lib/modules/login/controller/login_bloc.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import '../model/login_dto.dart';
+import '../repo/login_repo.dart';
+
+part 'login_event.dart';
+part 'login_state.dart';
+
+class LoginBloc extends Bloc<LoginEvent, LoginState> {
+  LoginBloc() : super(LoginState()) {
+    on<GoogleLogin>((event, emit) async {
+      emit(state.copyWith(
+        isLoading: true,
+      ));
+      final result = await LoginRepoImp().googleLogin();
+      final updatedState = result.fold(
+        (l) => state.copyWith(
+          isLoading: false,
+          errorMessage: l.response.toString(),
+          isUnauthenticated: true,
```

Comment:
Purpose of this **isUnauthenticated** ?

Reply:
When there is some error while performing google login, this state can be used.